storage/engine.go

```diff
@@ -662,6 +667,76 @@ func (e *Engine) deleteBucketRangeLocked(ctx context.Context, orgID, bucketID pl
 	return e.engine.DeletePrefixRange(ctx, name, min, max, pred)
 }
 
+func (e *Engine) CreateBackup(ctx context.Context) (int, []string, error) {
+	span, ctx := tracing.StartSpanFromContext(ctx)
+	span.LogKV("path", e.path)
+
+	err := e.engine.WriteSnapshot(ctx, tsm1.CacheStatusBackup)
+	if err != nil {
+		return 0, nil, err
+	}
+
+	id, snapshotPath, err := e.engine.FileStore.CreateSnapshot(ctx)
+	if err != nil {
+		return 0, nil, err
+	}
+
+	fileInfos, err := ioutil.ReadDir(snapshotPath)
+	if err != nil {
+		return 0, nil, err
+	}
+	filenames := make([]string, len(fileInfos))
+	for i, fi := range fileInfos {
+		filenames[i] = fi.Name()
+	}
+
+	return id, filenames, nil
+}
+
+func (e *Engine) FetchBackupFile(ctx context.Context, backupID int, backupFile string, w io.Writer) error {
+	span, ctx := tracing.StartSpanFromContext(ctx)
+	span.LogKV("path", e.path)
+
+	backupPath := e.engine.FileStore.InternalBackupPath(backupID)
+	if fi, err := os.Stat(backupPath); err != nil {
+		if os.IsNotExist(err) {
+			return errors.Errorf("backup %d not found", backupID)
+		}
+		return errors.WithMessagef(err, "failed to locate backup %d", backupID)
+	} else {
```

Comment:
Since all paths in the `if` lead to a `return` this `else` is redundant. Either make it an `else if` to save a line, or just have another `if` block.